commands/dota.js

```diff
@@ -0,0 +1,134 @@
+      break
+    default:
+      return sendHelp(channel)
+  }
+};
+
+async function getLatestMatch(client, channel, discordUser) {
+  const dotaIds = client.dotaConf
+  .filter(
+    entry => 
+      entry.channelId === channel.id && 
+      entry.discordId === discordUser.id)
+  .map(entry => entry.dotaId)
+  
+  if (!dotaIds.length) {
+    channel.send(`${discordUser.username} is not associated with a dota account`)
+    return
+  }
+
+  let latestStartTime = 0
+  let latestMatch = 0
+  for (const dotaId of dotaIds) {
+    let matchInfos = await dotaClient(`players/${dotaId}/recentMatches`)
+    for (const matchInfo of matchInfos) {
+      if (matchInfo.start_time > latestStartTime) {
+        latestStartTime = matchInfo.start_time
+        latestMatch = matchInfo
+      }
+    }
+  }
+
+  if (latestStartTime == 0) {
+    channel.send(`${discordUser.username} associated with dota accounts ${dotaIds} has not played a match recently`)
+  }
+  else {
+    const embedMsg = new discord.MessageEmbed()
+      .setColor('#0099ff')
+      .setTitle(`Match ${latestMatch.match_id}`)
+      .setDescription(`${discordUser.username} recently played in a ${formatSeconds(latestMatch.duration)} long match. 
+More info at [opendota](https://www.opendota.com/matches/${latestMatch.match_id}) [dotabuff](https://www.dotabuff.com/matches/${latestMatch.match_id})`)
+    channel.send(embedMsg)
+  }
+}
+
+async function registerLink(client, channel, discordUser, dotaId) {
+  const dotaInfo = await dotaClient(`players/${dotaId}`)
+      if (!validateId(dotaInfo)) {
+        channel.send(`could not find dota account for ${dotaId}`)
+      }
+      const dotaConfKey = getConfKey(channel.id, discordUser.id, dotaId)
+      client.dotaConf.set(dotaConfKey, {
+        dotaId: dotaId,
+        discordId: discordUser.id, 
+        channelId: channel.id
+      })
+
+      channel.send(`${discordUser.username} was linked to account ${dotaInfo.profile.personaname}`)
+}
+
+function formatSeconds(inputSeconds) {
+  const minutes = Math.floor(inputSeconds / 60)
+  const seconds = inputSeconds % 60
+  return `${minutes} ${minutes == 1 ? 'minute' : 'minutes'} and ${seconds} ${seconds == 1 ? 'second' : 'seconds'}`
+}
+
+/**
+ * Generates the dotaConf composite key from channelId, discordId, and dotaID
+ */
+function getConfKey(channelId, discordId, dotaId) {
+  return (`${channelId}-${discordId}-${dotaId}`)
+}
+
+function validateId(dotaInfo) {
+  return !!get(['profile', 'account_id'], dotaInfo)
+}
+
+function sendHelp(channel) {
+  return channel.send(`= USAGE =
+.dota register [dota account id]  :: Links your discordId to the specified dota account id for the current channel. Account id can be found in your dotabuff profile url (https://www.dotabuff.com/players/<ACCOUNT_ID>)
+.dota getLatestMatch :: gets the latest match for your linked dota account(s).`,
+  {code: 'asciidoc'});
+}
+
+function getDiscordUsername(client, discordId) {
```

Comment:
We should make this a util method, I had to do the same thing in the other command

Reply:
I've added this to functions.js as part of the client
usage (tested in debugger): 
```
>> client.getDiscordUsername(discordUser.id)
>>'trollpolice'
```